pkg/cmd/pipelinerun/delete.go

```diff
@@ -169,15 +188,20 @@ func pipelineRunLister(cs *cli.Clients, keep int, ns string) func(string) ([]str
 		if err != nil {
 			return nil, err
 		}
-		todelete, _ := keepPipelineRunsByNumber(pipelineRuns, keep)
+		var todelete []string
+		if since > 0 {
+			todelete, _ = keepPipelineRunsByAge(pipelineRuns, since)
+		} else {
```

Comment:
Do we need else here ??

Reply:
yes, this will check whether `--keep-since` flag passed or not. If not then deleteByNumber